### PR TITLE
chore: ARM ruby build

### PR DESCRIPTION
**Pull request by sighphyre**
This handles `feature_enabled` for Ruby and produces a valid build for MacOS+ARM and JRuby 21 (required for ARM MacOS, which our Ruby SDK currently enforces for _reasons_)

### CODE REVIEW

File: ruby-engine/build.sh [hunk 1]
```diff
@@ -1,4 +1,45 @@
+#!/bin/bash
```
Comment (sighphyre) [inline ruby-engine/build.sh:1]: You can mostly ignore this, it's a utility script so you don't have to manually copy the binaries into the Ruby folder for local dev

File: .github/workflows/publish-ruby.yaml [hunk 2]
```diff
@@ -1,8 +1,5 @@
 name: Build Ruby
 on:
```
Comment (sighphyre) [inline .github/workflows/publish-ruby.yaml:2]: The build now tags the binaries produces with both their platform (via the file extension) and their arch (new, in the filename)

File: unleash-yggdrasil/src/lib.rs [hunk 3]
```diff
@@ -650,6 +653,7 @@ impl VariantDef {
 }
 
 #[derive(Deserialize, Serialize, Debug, PartialEq, Eq, Clone)]
+#[serde(rename_all = "camelCase")]
```
Comment (sighphyre) [inline unleash-yggdrasil/src/lib.rs:656]: I checked Edge, this is never publicly exposed there so this should be safe

File: unleash-yggdrasil/src/strategy_upgrade.rs [hunk 4]
```diff
@@ -430,7 +430,7 @@ mod tests {
             variants: None,
```
Comment (sighphyre) [inline unleash-yggdrasil/src/strategy_upgrade.rs:430]: Nothing in this file is relevant, linter got an upgrade and it's decided to fix some tests

File: unleash-yggdrasil/src/lib.rs [hunk 5]
```diff
@@ -529,11 +529,12 @@ impl EngineState {
         context: &Context,
         external_values: &Option<HashMap<String, bool>>,
     ) -> Option<VariantDef> {
-        self.get_toggle(name).and_then(|toggle| {
+        self.get_toggle(name).map(|toggle| {
```
Comment (sighphyre) [inline unleash-yggdrasil/src/lib.rs:532]: This means that we now return the disabled variant if the feature exists but no variants are defined. This was previously not handled and was causing a spec test to fail (thank you client spec)

File: unleash-yggdrasil/src/lib.rs [hunk 6]
```diff
@@ -673,11 +675,11 @@ mod test {
     use serde::Deserialize;
     use std::{collections::HashMap, fs};
     use test_case::test_case;
-    use unleash_types::client_features::{ClientFeatures, FeatureDependency, Override};
+    use unleash_types::client_features::{ClientFeatures, FeatureDependency, Override, Payload};
```
Comment (sighphyre) [inline unleash-yggdrasil/src/lib.rs:678]: This and everything below in this file is just a consequence of having `feature_enabled` now properly camelCased on serialise/deserialise.

File: .gitignore [hunk 7]
```diff
@@ -20,7 +20,7 @@
 /go-engine/unleash_engine.h
 /php-engine/unleash_engine.h
 /php-engine/libyggdrasilffi.so
```
Comment (gastonfournier) [inline .gitignore:22]: Should we also change this for php-engine?

File: ruby-engine/build.sh (hunk 1 above)
Comment (gastonfournier) [reply ruby-engine/build.sh:1]: I would set it to fail if something goes wrong
```suggestion
#!/bin/bash
set -e
```

**Review by gastonfournier — APPROVED** [review]
Looks sane to me! Just some minor comments that can be dismissed

File: ruby-engine/build.sh (hunk 1 above)
Comment (sighphyre) [reply ruby-engine/build.sh:1]: Thanks! Good suggestion!

File: .gitignore (hunk 7 above)
Comment (sighphyre) [reply .gitignore:22]: Honestly I'm not sure yet. I'm don't yet understand how PHP distributes native binaries or whether or not these tricks will be needed there but definitely in a future PR

**Pull request merged**